### DISCUSSION TITLE
Found a mistake on EigenProblem.h

### DISCUSSION THREAD

**Discussion by doquang** [idaholab/moose#22622] [Q&A Getting Started] [answered]
In the rev 6ff686419915623806e9a0b5d3155e56d6cf4b7e, @lindsayad added a new method getNonlinearEigenSystem(unsigned int nl_sys_num = 0) into framework/include/problems/EigenProblem.h. 
The new method is inside a condition #ifdef LIBMESH_HAVE_SLEPC, but its implementation is outside this condition. It prevents to compile on a system that do not have SLEPC.

**Comment by lindsayad** [comment, marked as answer]
Thank you for the report @doquang . I will create an issue for this

**Reply by lindsayad** [reply]
Opened #22628

**Reply by GiudGiud** [reply]
Issue is fixed
